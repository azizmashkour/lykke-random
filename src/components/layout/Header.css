.header-title {
  background: rgb(0, 150, 136);
  color: #fff;
  font-size: 2rem;
  font-weight: 900;
  height: 4rem;
  line-height: 1.7;
  position: fixed;
  width: 100%;
  z-index: 9;
}
