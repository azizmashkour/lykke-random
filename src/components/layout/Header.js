import React, { Fragment } from 'react';
import logo from './logo.svg';
import './Header.css';

const Header = () => (
  <Fragment>
    <div className="header-title">
      <span>
        Discover random images with a filter option
      </span>
    </div>
    <header className="app-header">
      <img src={logo} className="app-logo" alt="logo" />
      <p>
        Filter images by related categories. Add&nbsp;
        <code>?mode=forest</code>
        &nbsp; in the URL params for forest images.
      </p>
      <a
        className="app-link"
        href="https://reactjs.org"
        target="_blank"
        rel="noopener noreferrer"
      >
        Learn React
      </a>
    </header>
  </Fragment>
);

export default Header;
