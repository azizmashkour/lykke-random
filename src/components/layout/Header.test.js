import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe('render header component tests', () => {
  const component = shallow(<Header />);

  it('has one instance of header', () => {
    expect(component).toMatchSnapshot();
  });

  it('has one <div /> with class header-title', () => {
    expect(component.find('div.header-title').length).toEqual(1);
  });

  it('has one <header /> with class app-header', () => {
    expect(component.find('header.app-header').length).toEqual(1);
  });

  it('has one <img /> with class app-logo', () => {
    expect(component.find('img.app-logo').length).toEqual(1);
  });

  it('has one <a /> with class app-link', () => {
    expect(component.find('a.app-link').length).toEqual(1);
  });
});
