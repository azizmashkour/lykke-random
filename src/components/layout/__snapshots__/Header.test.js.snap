// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render header component tests has one instance of header 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="header-title"
        >
          <span>
            Discover random images with a filter option
          </span>
        </div>,
        <header
          className="app-header"
        >
          <img
            alt="logo"
            className="app-logo"
            src="logo.svg"
          />
          <p>
            Filter images by related categories. Add 
            <code>
              ?mode=forest
            </code>
              in the URL params for forest images.
          </p>
          <a
            className="app-link"
            href="https://reactjs.org"
            rel="noopener noreferrer"
            target="_blank"
          >
            Learn React
          </a>
        </header>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span>
            Discover random images with a filter option
          </span>,
          "className": "header-title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Discover random images with a filter option",
          },
          "ref": null,
          "rendered": "Discover random images with a filter option",
          "type": "span",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="logo"
              className="app-logo"
              src="logo.svg"
            />,
            <p>
              Filter images by related categories. Add 
              <code>
                ?mode=forest
              </code>
                in the URL params for forest images.
            </p>,
            <a
              className="app-link"
              href="https://reactjs.org"
              rel="noopener noreferrer"
              target="_blank"
            >
              Learn React
            </a>,
          ],
          "className": "app-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "app-logo",
              "src": "logo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Filter images by related categories. Add ",
                <code>
                  ?mode=forest
                </code>,
                "  in the URL params for forest images.",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Filter images by related categories. Add ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "?mode=forest",
                },
                "ref": null,
                "rendered": "?mode=forest",
                "type": "code",
              },
              "  in the URL params for forest images.",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Learn React",
              "className": "app-link",
              "href": "https://reactjs.org",
              "rel": "noopener noreferrer",
              "target": "_blank",
            },
            "ref": null,
            "rendered": "Learn React",
            "type": "a",
          },
        ],
        "type": "header",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="header-title"
          >
            <span>
              Discover random images with a filter option
            </span>
          </div>,
          <header
            className="app-header"
          >
            <img
              alt="logo"
              className="app-logo"
              src="logo.svg"
            />
            <p>
              Filter images by related categories. Add 
              <code>
                ?mode=forest
              </code>
                in the URL params for forest images.
            </p>
            <a
              className="app-link"
              href="https://reactjs.org"
              rel="noopener noreferrer"
              target="_blank"
            >
              Learn React
            </a>
          </header>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span>
              Discover random images with a filter option
            </span>,
            "className": "header-title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Discover random images with a filter option",
            },
            "ref": null,
            "rendered": "Discover random images with a filter option",
            "type": "span",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="logo"
                className="app-logo"
                src="logo.svg"
              />,
              <p>
                Filter images by related categories. Add 
                <code>
                  ?mode=forest
                </code>
                  in the URL params for forest images.
              </p>,
              <a
                className="app-link"
                href="https://reactjs.org"
                rel="noopener noreferrer"
                target="_blank"
              >
                Learn React
              </a>,
            ],
            "className": "app-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": "app-logo",
                "src": "logo.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Filter images by related categories. Add ",
                  <code>
                    ?mode=forest
                  </code>,
                  "  in the URL params for forest images.",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Filter images by related categories. Add ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "?mode=forest",
                  },
                  "ref": null,
                  "rendered": "?mode=forest",
                  "type": "code",
                },
                "  in the URL params for forest images.",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Learn React",
                "className": "app-link",
                "href": "https://reactjs.org",
                "rel": "noopener noreferrer",
                "target": "_blank",
              },
              "ref": null,
              "rendered": "Learn React",
              "type": "a",
            },
          ],
          "type": "header",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
